include/mapbox/variant.hpp

```diff
@@ -84,25 +84,27 @@ struct MAPBOX_VARIANT_DEPRECATED static_visitor
     ~static_visitor() {}
 };
 
+using type_index_t = unsigned int;
```

Comment:
Is an `unsigned` really needed? Shouldn't a `uint8_t` be good enough as outlined in the ticket?

Also should we `static_assert` that users never use more types we can index?